clients/segments/segment_test.go

```diff
@@ -460,7 +466,13 @@ func TestUpsert(t *testing.T) {
 func TestDelete(t *testing.T) {
 	t.Run("call without ID returns an error", func(t *testing.T) {
```

Comment:
I would argue that this is testing of API client functionality. So how about moving to `api/clients/segments` ?